flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
+            )
+            return
+
+    def _add_forcing_rain(self, forcing: IRainfall):
+        """Add spatially constant rain forcing to sfincs model. Use timeseries or a constant magnitude.
+
+        Parameters
+        ----------
+        timeseries : Union[str, os.PathLike], optional
+            path to file of timeseries file (.csv) which has two columns: time and precipitation, by default None
+        const_intensity : float, optional
+            time-invariant precipitation intensity [mm/hr], by default None
+        """
+        if isinstance(forcing, RainfallConstant):
+            self._model.setup_precip_forcing(
+                timeseries=None,
+                magnitude=forcing.intensity,
+            )
+        elif isinstance(forcing, RainfallSynthetic):
+            self._model.setup_precip_forcing(
+                timeseries=forcing.path,
+                magnitude=None,
+            )
+        elif isinstance(forcing, RainfallFromModel):
+            # TODO: check how to add this to the model
+            data = self._get_wl_df_from_offshore_his_results()
+            self._model.setup_precip_forcing_from_grid(precip=data)
+
+        elif isinstance(forcing, RainfallFromSPWFile):
+            # TODO: check how to add this to the model
+            pass
+        else:
+            self._logger.warning(
+                f"Unsupported rainfall forcing type: {forcing.__class__.__name__}"
+            )
+            return
+
+    def _add_forcing_discharge(self, forcing: IDischarge):
+        """Add spatially constant discharge forcing to sfincs model. Use timeseries or a constant magnitude.
+
+        Parameters
+        ----------
+        timeseries : Union[str, os.PathLike], optional
+            path to file of timeseries file (.csv) which has two columns: time and discharge, by default None
+        const_discharge : float, optional
+            time-invariant discharge magnitude [m3/s], by default None
+        """
+        # TODO investigate where to include rivers from site.toml
+        if isinstance(forcing, DischargeConstant):
+            self._model.setup_discharge_forcing(
+                timeseries=None,
+                magnitude=forcing.discharge,
+            )
+        elif isinstance(forcing, DischargeSynthetic):
+            self._model.setup_discharge_forcing(
+                timeseries=forcing.file,
+                magnitude=None,
+            )
+        else:
+            self._logger.warning(
+                f"Unsupported discharge forcing type: {forcing.__class__.__name__}"
+            )
+            return
+
+    def _add_forcing_waterlevels(self, forcing: IWaterlevel):
+        if isinstance(forcing, WaterlevelSynthetic):
+            # TODO implement this
+            pass
+        elif isinstance(forcing, WaterlevelFromFile):
+            # TODO implement this
+            pass
+        elif isinstance(forcing, WaterlevelFromModel):
+            # TODO implement this
```

Comment:
This is where we need to read this from the offshore model (for historic hurricane and historic offshore).  This is where we need the function self._get_wl_df_from_offshore_his_results()

Reply:
I see, Im not sure what will be better/nicer:
- calling the function right here and passing the data directly
- calling the function in event.process() and then storing the data in a csv file, which is then read here